src/rockstor/storageadmin/static/storageadmin/js/views/dashboard/pool_usage.js

```diff
@@ -69,144 +78,73 @@ PoolUsageWidget = RockStorWidgetView.extend({
-        .attr('y', 0)
-        .attr('rx', 4)
-        .attr('ry', 4)
-        .attr('width', function(d) { return _this.x(d.get('percentUsed')); })
-        .attr('height', this.barHeight - 2);
-
-        // Pool Name
-        dataRow.append("text")
-        .attr('class', 'share-name')
-        .attr('x', 45 + this.barWidth)
-        .attr('y', this.textOffset)
-        .style("text-anchor", "end")
-        .text(function(d) {
-            var n = d.get('name');
-            // truncate name to 15 chars
-            if (n.length > 15) {
-                n = n.slice(0,12) + '...';
-            }
-            return n + ' (' + humanize.filesize(d.get('bytesUsed') * 1024) +
-                '/' + humanize.filesize(d.get('size') * 1024) +
-                ')' ;
-        })
-        .on('mouseover', tip.show)
-        .on('mouseout', tip.hide);
-
-        // Legend
-        var legend = this.svg.append('g')
-        .attr('class', 'legend')
-        .attr("transform", function(d, i) {
-            return "translate(0," + ((_this.data.length + 1) * _this.rowHeight) + ")";
+        this.$('.pused').each(function(index) {
+            $(this).text(_this.data[index].get('percentUsed').toFixed(2) + '%');
+        });
+        var truncate = _this.maximized ? 100 : 12;
+        this.$('.progress-animate').each(function(index) {
+            $(this).find('span')
+                .text(humanize.truncatechars(_this.data[index].get('name'), truncate) +
+                    '(' + humanize.filesize(_this.data[index].get('bytesUsed') * 1024) +
+                    '/' + humanize.filesize(_this.data[index].get('bytesFree') * 1024) +
+                    ')');
+            $(this).animate({
+                width: _this.data[index].get('percentUsed').toFixed(2) + '%'
+            }, 1000);
         });
+    },
 
-        legend.append('rect')
-        .attr('x', 50)
-        .attr('y', 10)
-        .attr('width', 10)
-        .attr('height', 10)
-        .attr('class', 'bar used');
-
-        legend.append('text')
-        .attr('x', 50 + 10 + 2)
-        .attr('y', 20)
-        .style("text-anchor", "start")
-        .text('Used');
-
-        legend.append('rect')
-        .attr('x', 50 + 50)
-        .attr('y', 10)
-        .attr('width', 10)
-        .attr('height', 10)
-        .attr('class', 'bar unused');
-
-        legend.append('text')
-        .attr('x', 50 + 50 + 10 + 2)
-        .attr('y', 20)
-        .style("text-anchor", "start")
-        .text('Free');
+    buildProgressbar: function() {
 
+        var _this = this;
+        var percent_div = {
+            class: 'pused',
+            style: 'font-size: 10px; text-align: right; padding-right: 5px; display: table-cell; width: 50px; vertical-align: middle;'
+        };
+        var progressbar_container = {
+            class: 'progress',
+            style: 'display: inline-block; margin: 0px; position: relative; height: ' + _this.barHeight + 'px; width: ' + _this.barWidth + 'px;'
```

Comment:
Ok class styling for some elements, but not all of these are "static" (consider that _this.barHeight/Width for dynamic progressbar dimensions), so having it in a class would aniway require some inline style

Flyer